tdrs-backend/tdpservice/users/views.py

```diff
@@ -28,7 +29,7 @@ class UserViewSet(
 ):
     """User accounts viewset."""
 
-    permission_classes = [UserPermissions]
+    permission_classes = [IsAuthenticated, UserPermissions]
```

Comment:
Without this the `set_profile` action encounters a 500 error for anonymous requests. This will instead ensure they get a 401 response as expected.